File: scripts/set_webhook.py
"""Register the Telegram webhook. Run once after deploying.

Usage: uv run python scripts/set_webhook.py <APP_URL>
Example: uv run python scripts/set_webhook.py https://rkbot-xyz.koyeb.app
"""
import sys
import os
import httpx


def main():
    if len(sys.argv) != 2:
        print(f"Usage: {sys.argv[0]} <APP_URL>")
        sys.exit(1)

    app_url = sys.argv[1].rstrip("/")
    token = os.environ.get("TELEGRAM_BOT_TOKEN")
    secret = os.environ.get("WEBHOOK_SECRET")

    if not token or not secret:
        print("Error: Set TELEGRAM_BOT_TOKEN and WEBHOOK_SECRET env vars")
        sys.exit(1)

    response = httpx.post(
        f"https://api.telegram.org/bot{token}/setWebhook",
        json={
            "url": f"{app_url}/webhook",
            "secret_token": secret,
            "allowed_updates": ["message"],
        },
    )

    result = response.json()
    if result.get("ok"):
        print(f"Webhook set to {app_url}/webhook")
    else:
        print(f"Error: {result}")
        sys.exit(1)


if __name__ == "__main__":
    main()

File: scripts/set-webhook.ts
/**
 * Register the Telegram webhook endpoint.
 *
 * Usage:
 *   npx tsx scripts/set-webhook.ts <RENDER_URL>
 *
 * Example:
 *   npx tsx scripts/set-webhook.ts https://rk-bot.onrender.com
 *
 * Reads TELEGRAM_BOT_TOKEN and TELEGRAM_WEBHOOK_SECRET from .env or environment.
 */

const url = process.argv[2];
if (!url) {
  console.error("Usage: npx tsx scripts/set-webhook.ts <BASE_URL>");
  console.error("Example: npx tsx scripts/set-webhook.ts https://rk-bot.onrender.com");
  process.exit(1);
}

const token = process.env.TELEGRAM_BOT_TOKEN;
const secret = process.env.TELEGRAM_WEBHOOK_SECRET;

if (!token) {
  console.error("Missing TELEGRAM_BOT_TOKEN env var");
  process.exit(1);
}
if (!secret) {
  console.error("Missing TELEGRAM_WEBHOOK_SECRET env var");
  process.exit(1);
}

const webhookUrl = `${url.replace(/\/$/, "")}/webhook`;

const res = await fetch(`https://api.telegram.org/bot${token}/setWebhook`, {
  method: "POST",
  headers: { "Content-Type": "application/json" },
  body: JSON.stringify({
    url: webhookUrl,
    secret_token: secret,
    allowed_updates: ["message", "callback_query"],
  }),
});

const data = await res.json();

if (data.ok) {
  console.log(`Webhook set to: ${webhookUrl}`);
  console.log("Response:", JSON.stringify(data, null, 2));
} else {
  console.error("Failed to set webhook:", JSON.stringify(data, null, 2));
  process.exit(1);
}
